ios/brave-ios/Sources/AIChat/Subscription/SDK/BraveStoreSDK.swift

```diff
@@ -0,0 +1,412 @@
+  private(set) var leoSubscriptionStatus: Product.SubscriptionInfo.Status?
+  
+  // MARK: - Private
+  
+  /// All observers this uses
+  private var observers = [AnyCancellable]()
+  
+  private override init() {
+    super.init()
+
+    // Observe Product Updates
+    observers.append($allProducts.sink(receiveValue: onProductsUpdated(_:)))
+    
+    // Observe Customer Purchase Updates
+    observers.append($purchasedProducts.sink(receiveValue: onPurchasesUpdated(_:)))
+  }
+  
+  // MARK: - Public
+  
+  /// A list of all products the Brave Store offers
+  public override var allAppStoreProducts: [any AppStoreProduct] {
+    return BraveStoreProduct.allCases
+  }
+  
+  /// The current store environment
+  public var enviroment: BraveStoreEnvironment {
+    // Retrieve the subscription renewal information
+    guard let renewalInfo = [vpnSubscriptionStatus, leoSubscriptionStatus].compactMap({ $0 }).first?.renewalInfo else {
+      // There is currently no subscription so check if there was a restored receipt
+      if Bundle.main.appStoreReceiptURL?.lastPathComponent == "sandboxReceipt" {
+        return .sandbox
+      }
+      
+      // If the application has a Parent Process-ID then it is being debugged
+      if getppid() != 1 {
+        return .xcode
+      }
+      
+      return .production
+    }
+    
+    // Retrieve the current environment from StoreKit's Renewal Information
+    switch renewalInfo {
+    case .verified(let renewalInfo), .unverified(let renewalInfo, _):
+      if #available(iOS 16.0, *) {
+        return .init(rawValue: renewalInfo.environment.rawValue) ?? .production
+      }
+      
+      return .init(rawValue: renewalInfo.environmentStringRepresentation) ?? .production
+    }
+  }
+  
+  /// A boolean indicating whether or not all the VPN product offerings has been loaded
+  public var isVpnProductsLoaded: Bool {
+    if vpnMonthlyProduct != nil || vpnYearlyProduct != nil {
+      return true
+    }
+    
+    return false
+  }
+  
+  /// A boolean indicating whether or not all the Leo product offerings has been loaded
+  public var isLeoProductsLoaded: Bool {
+    if leoMonthlyProduct != nil || leoYearlyProduct != nil {
+      return true
+    }
+    
+    return false
+  }
+  
+  /// Refreshes all the Skus SDK orders
+  public func refreshAllSkusOrders() {
+    
+  }
+  
+  /// Restores a single purchased product
+  /// - Parameter product: The product whose purchase receipt to restore
+  /// - Returns: Returns true if receipt restoration was successful. False otherwise
+  public func restorePurchase(_ product: BraveStoreProduct) async -> Bool {
+    if await currentTransaction(for: product) != nil {
+      try? await AppStoreReceipt.sync()
+      return true
+    }
+    
+    return false
+  }
+  
+  /// Restores all purchased products
+  /// - Returns: Returns true if receipt restoration was successful. False otherwise
+  @MainActor
+  public func restorePurchases() async -> Bool {
+    #if STOREKIT2
```

Comment:
Where are these conditional compilation flags defined? I don't see any changes to the xcconfig's

Reply:
I added this code to future proof the SDK. The backend does not currently support this part YET.
This is extremely useful when testing in the simulator as the old API doesn't let subscriptions load in the simulator, but this one does.

Reply:
Gotcha, maybe toss a comment in there explaining this then